subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/DefaultArtifactHandler.java

```diff
@@ -63,8 +65,8 @@ private PublishArtifact pushArtifact(Configuration configuration, Object notatio
 
     private void warnIfConfigurationIsDeprecated(DeprecatableConfiguration configuration) {
         if (configuration.isFullyDeprecated()) {
-            DeprecationLogger.nagUserOfReplacedConfiguration(configuration.getName(), DeprecationLogger.ConfigurationDeprecationType.ARTIFACT_DECLARATION,
-                GUtil.flattenElements(configuration.getDeclarationAlternatives(), configuration.getConsumptionAlternatives()));
+            DeprecationLogger.nagUserWith(DeprecationMessage.replacedConfiguration(configuration.getName(), ConfigurationDeprecationType.ARTIFACT_DECLARATION,
```

Comment:
This is another good example of where a specialized builder would come in handy.  Internally, it could be represented as an enum, but a specialized "deprecated configuration builder" would be able to hide that.